packages/components/src/drop-zone/provider.js

```diff
@@ -200,21 +244,23 @@ class DropZoneProvider extends Component {
 		event.dataTransfer && event.dataTransfer.files.length; // eslint-disable-line no-unused-expressions
 
 		const { position, hoveredDropZone } = this.state;
-		const dragEventType = getDragEventType( event );
+		const normalizedDragEvent = this.getNormalizedDragEvent( event );
 		const dropZone = this.dropZones[ hoveredDropZone ];
 		this.resetDragState();
 
 		if ( dropZone ) {
-			switch ( dragEventType ) {
+			switch ( normalizedDragEvent.type ) {
 				case 'file':
-					dropZone.onFilesDrop( [ ...event.dataTransfer.files ], position );
+					dropZone.onFilesDrop( [ ...normalizedDragEvent.files ], position );
 					break;
 				case 'html':
-					dropZone.onHTMLDrop( event.dataTransfer.getData( 'text/html' ), position );
+					dropZone.onHTMLDrop( normalizedDragEvent.html, position );
 					break;
 				case 'default':
 					dropZone.onDrop( event, position );
 			}
+
+			dropZone.onUniversalDrop( normalizedDragEvent, position );
```

Comment:
This is technically not useful in this PR, but I find the three different callbacks per data type to be a bad API. this tries to introduce a non-breaking change API to replace all three callbacks with a normalized callback equivalent to the newly introduced onDragOver one.